Mathlib/Topology/Algebra/Module/LocallyConvex.lean

```diff
@@ -192,3 +192,40 @@ instance Prod.locallyConvexSpace [TopologicalSpace E] [TopologicalSpace F] [Loca
     (locallyConvexSpace_induced (LinearMap.snd _ _ _))
 
 end LatticeOps
+
+section LinearOrderedSemiring
+
+instance {R : Type*} [TopologicalSpace R] [LinearOrderedSemiring R] [OrderTopology R] :
+    LocallyConvexSpace R R where
+  convex_basis x := by
+    obtain hl | hl := isBot_or_exists_lt x
+    · let _ : Bot R := ⟨x⟩
+      let _ : OrderBot R := ⟨hl⟩
+      replace hu : x = ⊥ := rfl
+      rw [hu]
+      refine nhds_bot_basis.to_hasBasis' ?_ ?_
```

Comment:
I think there's some API missing here in terms of `IsBot`; perhaps an induction principle like:
```

@[elab_as_elim]
def IsBot.rec {α} [LE α] {P : (x : α) → IsBot x → Sort*} (h : ∀ [OrderBot α], P ⊥ isBot_bot) (x : α) (hx : IsBot x) : P x hx := by
  letI : OrderBot α := { bot := x, bot_le := hx }
  apply h
```

Thoughts, @YaelDillies?

Reply:
yes that would make sense